### DISCUSSION TITLE
Recompiling SeisSol on supermuc ng

### DISCUSSION THREAD

**Discussion by jrekoske** [SeisSol/SeisSol#532] [Q&A] [answered]
Hi, I'm trying to recompile SeisSol on supermuc ng with the updated modules from #529. As I'm following the documentation, when I install easi, I get the following error:

`CMake Error at CMakeLists.txt:36 (find_package):
  By not providing "Findyaml-cpp.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "yaml-cpp",
  but CMake did not find one.`

  `Could not find a package configuration file provided by "yaml-cpp"
  (requested version 0.6) with any of the following names:
   yaml-cppConfig.cmake
   yaml-cpp-config.cmake`

  `Add the installation prefix of "yaml-cpp" to CMAKE_PREFIX_PATH or set
  "yaml-cpp_DIR" to a directory containing one of the above files.  If
  "yaml-cpp" provides a separate development package or SDK, be sure it has
  been installed.`

I have loaded the module `yaml-cpp/0.7.0-intel21` and there doesn't appear to be a yaml-cpp version 0.6 with the new spack stack.

**Comment by sebwolf-de** [comment, marked as answer]
Did you remove everything from the build folder? It might be the case, that you load the new module, but there is still some old path in the CmakeCache.

**Reply by jrekoske** [reply]
Thanks, I now have easi installed succesfully but the SeisSol configuration step still gives an error:

CMake Error at CMakeLists.txt:175 (find_package):
  Found package configuration file:

    /dss/dsshome1/0B/di46bak/easi_build/easiConfig.cmake

  but it set easi_FOUND to FALSE so package "easi" is considered to be NOT
  FOUND.  Reason given by package:

  easi could not be found because dependency OpenMP could not be found.

**Reply by krenzland** [reply]
> easi could not be found because dependency OpenMP could not be found.

Are you sure that you've used the same compiler for both easi and SeisSol? I can't see any reason why openmp might not be found on SNG.

**Reply by jrekoske** [reply]
Hi, I'm attempting a new (clean) installation and cannot install easi due to the same error I posted originally (yaml-cpp version 0.6 not found). I have checked that I have removed everything from the build folder.

The documentation for the easi installation is also innacurate because "module load intel mpi-intel" and "module load yaml-cpp/0.6.3-intel19" are no longer available.

**Reply by krenzland** [reply]
I cannot reproduce this. I'd recommend to build yaml-cpp from source then, as outlined in the easi documentation.

**Reply by krenzland** [reply]
We've updated the easi documentation with PR https://github.com/SeisSol/easi/pull/34

**Reply by Thomas-Ulrich** [reply]
please check : https://github.com/SeisSol/SeisSol/pull/539/files

**Reply by jrekoske** [reply]
I followed the newly updated documentation and using the pre-compiled libraries I obtain errors at [this](https://seissol.readthedocs.io/en/latest/supermuc.html#:~:text=bin/pspamm.py%20..-,make%20%2Dj%2048,-Note%20that%20to) step:

`[ 99%] Linking CXX executable SeisSol_Release_dskx_4_elastic`
`ld: cannot find -lpnetcdf`
`ld: cannot find -lhdf5_hl`
`ld: cannot find -lhdf5`
`make[2]: *** [CMakeFiles/SeisSol-bin.dir/build.make:350: SeisSol_Release_dskx_4_elastic] Error 1`
`make[1]: *** [CMakeFiles/Makefile2:146: CMakeFiles/SeisSol-bin.dir/all] Error 2`
`make[1]: *** Waiting for unfinished jobs....`
`[100%] Linking CXX executable SeisSol_proxy_Release_dskx_4_elastic`
`ld: cannot find -lpnetcdf`
`ld: cannot find -lhdf5_hl`
`ld: cannot find -lhdf5`
`make[2]: *** [CMakeFiles/SeisSol-proxy.dir/build.make:351: SeisSol_proxy_Release_dskx_4_elastic] Error 1`
`make[1]: *** [CMakeFiles/Makefile2:94: CMakeFiles/SeisSol-proxy.dir/all] Error 2`
`make: *** [Makefile:91: all] Error 2`

Is this related to Thomas' comment about fixing hdf5 in CMakelist?

**Reply by Thomas-Ulrich** [reply]
My understanding is that cmake still use some of your compiled libraries, probably because in your ~/.bashrc file, you still have things like:
```
export PATH=~/bin/:$PATH
export PKG_CONFIG_PATH=~/lib/pkgconfig:$PKG_CONFIG_PATH
export LD_LIBRARY_PATH=~/lib:$LD_LIBRARY_PATH
```
I would suggest to remove ~/bin ~/include and ~/lib.
As a workaround, If it does not work you can manually add the link to the correct hdf5 in the CMakeList:

```
link_directories(/dss/dsshome1/lrz/sys/spack/release/22.2.1/opt/skylake_avx512/netcdf-hdf5-all/4.7_hdf5-1.10-intel-ymxvq7k/lib/)
```
e.g. after
```
target_link_libraries(SeisSol-lib PUBLIC ${HDF5_C_HL_LIBRARIES} ${HDF5_C_LIBRARIES})
```
(but that should not be necessary)

**Reply by jrekoske** [reply]
I had already removed those three lines from my .bashrc, but I tried removing the ~/bin, ~/include, and ~/lib folders but that did not fix it.

Adding the `link_directories` line to the CMakeList seemed to do the trick.

**Reply by Thomas-Ulrich** [reply]
well, I recompiled again and got the same problem as you... very strange. Anyway, now you have it compiled ;-)

**Reply by Shihao-Yuan** [reply]
Add $NETCDF_HDF5_ALL_BASE/lib/ again to LIBRARY_PATH
